src/test/java/org/generationcp/middleware/api/program/ProgramServiceImplTest.java

```diff
@@ -0,0 +1,51 @@
+package org.generationcp.middleware.api.program;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.generationcp.middleware.IntegrationTestBase;
+import org.generationcp.middleware.manager.WorkbenchDaoFactory;
+import org.generationcp.middleware.pojos.workbench.CropType;
+import org.generationcp.middleware.pojos.workbench.Project;
+import org.generationcp.middleware.pojos.workbench.WorkbenchUser;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Date;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+
+public class ProgramServiceImplTest extends IntegrationTestBase {
+
+	private ProgramServiceImpl programService;
+
+	private Project testProject;
+
+	private WorkbenchUser workbenchUser;
+
+	@Before
+	public void setUp() {
+		this.programService = new ProgramServiceImpl(this.workbenchSessionProvider);
+		final WorkbenchDaoFactory daoFactory = new WorkbenchDaoFactory(this.workbenchSessionProvider);
+
+		if (this.testProject == null) {
+			this.testProject = new Project();
+			this.testProject.setProjectName("Project " + RandomStringUtils.randomAlphanumeric(10));
+			this.testProject.setStartDate(new Date());
+			this.testProject.setUniqueID(UUID.randomUUID().toString());
+			this.testProject.setLastOpenDate(new Date());
+			this.testProject.setCropType(new CropType("maize"));
+			daoFactory.getProjectDAO().save(this.testProject);
+		}
+
+		final Integer userId = this.findAdminUser();
+		this.workbenchUser = daoFactory.getWorkbenchUserDAO().getById(userId);
+	}
+
+	@Test
+	public void saveOrUpdateProjectUserInfo() {
```

Comment:
This test is only testing saving the last used project, it is not testing the update (IF condition) @cuenyad